components/AdminUserSettings/AdminUserSettings.tsx

```diff
@@ -1,25 +1,75 @@
+import React, { JSX } from 'react';
+import { Button } from '../Button/Button';
+import { useDataStore } from '@/store/dataStore';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '../TableDropDownMenu/TableDropDownMenu';
 import { LucideChevronsUpDown } from 'lucide-react';
-import React from 'react';
+import { useAuthContext } from '@/context/AuthContextProvider';
+import { useRouter } from 'next/navigation';
+import LinkCustom from '../LinkCustom/LinkCustom';
 
 /**
- * The admin user settings component.
- * @returns The rendered admin user settings.
+ * Renders admin user settings.
+ * @returns {JSX.Element} The rendered admin user settings component.
  */
-export const AdminUserSettings = (): React.JSX.Element => {
+export const AdminUserSettings = (): JSX.Element => {
+  const router = useRouter();
+  const { logoutAccount } = useAuthContext();
+  const { user } = useDataStore((state) => state);
+
+  /**
+   * Handles the logout.
+   * @returns {Promise<void>} The logout promise.
+   */
+  const handleLogout = async (): Promise<void> => {
+    try {
+      await logoutAccount();
+      router.push('/login');
+    } catch (error) {
+      throw new Error('Logout failed');
+    }
+  };
+
   return (
-    <div
-      className="admin-user-settings flex gap-2 px-2 py-2 items-center outline outline-border rounded text-foreground"
-      data-testid="admin-user-settings"
-    >
-      <span className="bg-cyan-500 w-8 h-8 rounded-full" />
-      <p>Users Name</p>
-      <LucideChevronsUpDown
-        className="ml-auto text-muted-foreground"
-        size={16}
-      />
-    </div>
+    <DropdownMenu>
+      <DropdownMenuTrigger
+        className="w-56 focus:outline-none mx-auto"
+        data-testid="admin-user-settings"
+      >
+        <div className="admin-user-settings w-full flex space-between gap-2 px-2 py-2 items-center border border-border rounded-lg text-foreground overflow-hidden">
+          <span className="bg-cyan-500 w-8 h-8 rounded-full" />
+          <p className="truncate ... w-36 ">{user.email}</p>
+          <LucideChevronsUpDown className="text-muted-foreground" size={16} />
+        </div>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="border border-border rounded-lg p-0 w-56 ">
+        <DropdownMenuItem className="cursor-pointer rounded-b-none flex focus:bg-muted">
+          <LinkCustom
+            className="text-base no-underline hover:text-foreground w-full py-2 px-0 text-muted-foreground hover:underline"
+            href="/account/settings"
+            data-testid="edit-profile-link"
+          >
+            Edit Profile
+          </LinkCustom>
+        </DropdownMenuItem>
+        <DropdownMenuItem className="cursor-pointer rounded-none focus:bg-muted">
+          <Button
+            className="w-full text-base no-underline py-2 px-0 h-auto text-muted-foreground hover:text-foreground font-normal justify-normal hover:underline"
```

Comment:
Is this a special button that it needs all these different class names? @ryandotfurrer Can we create a new variant for this type of button so we're not setting these classes over again?

Reply:
@shashilo By default, shadcn uses a different look that I needed to change to match our design (first screenshot). Shadcn's does not take up the full width (second screenshot).
---
![CleanShot 2024-11-14 at 20 57 33@2x](https://github.com/user-attachments/assets/07188aa3-b78a-42d5-85af-a816942cc870)
Figma design
---
![CleanShot 2024-11-14 at 20 53 30@2x](https://github.com/user-attachments/assets/e9f6ad48-ca85-47ae-94fd-8cb7083f0bfe)
shadcn default

Reply:
Let's do that, but in another ticket.